### PR TITLE
python3Packages.black: init at 18.4a0

### PR DESCRIPTION
**Pull request by sveitser**
###### Motivation for this change
Add python code formatter `black`: https://github.com/ambv/black.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/black/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, buildPythonPackage, fetchPypi, pythonOlder, attrs, click }:
+
+buildPythonPackage rec {
+  pname = "black";
+  version = "18.4a0";
+  name = "${pname}-${version}";
```
Comment (adisbladis) [inline pkgs/development/python-modules/black/default.nix]: When using `buildPythonPackage` the `name` attribute defaults to `name = "${pname}-${version}";` so this line can be omitted.

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/black/default.nix (hunk 1 above)
Comment (sveitser) [reply pkgs/development/python-modules/black/default.nix]: Good to know. Thanks!

File: pkgs/development/python-modules/black/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "04dffr4wmzs4vf2xj0cxp03hv04x0kk06qyzx6jjrp1mq0z3n2rr";
+  };
+
+  propagatedBuildInputs = [ attrs click ];
+
+  meta = with stdenv.lib; {
+    description = "The uncompromising Python code formatter.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/black/default.nix]: No period: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/python-modules/black/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    homepage    = "https://github.com/ambv/black";
```
Comment (dotlambda) [inline pkgs/development/python-modules/black/default.nix]: no quotes

**Review by dotlambda — CHANGES_REQUESTED** [review]
If I understand correctly, `black` doesn't provide a python module. In that case, it must be moved to e.g. `pkgs/development/tools/black` and called from `all-packages.nix`.

**Comment by sveitser** [comment]
@dotlambda made the changes for meta attributes.

It can also be used as a python module, for instance here https://github.com/ambv/black/blob/master/vim/plugin/black.vim#L77

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.black

**Pull request merged**